message_tagging_service/tagging_service.py

```diff
@@ -376,11 +364,36 @@ def tag_build(nvr, dest_tags, koji_session):
 
 
 def handle(rule_defs, event_msg):
+    """Handle MBS build.state.change event"""
+
     this_name = event_msg["name"]
     this_stream = event_msg["stream"]
     this_version = event_msg["version"]
     this_context = event_msg["context"]
     nsvc = f"{this_name}-{this_stream}-{this_version}-{this_context}"
+    state_name = event_msg['state_name']
+
+    # Rule definitions are grouped by build state:
+    # ready: [{...}, {...}]
+    # done: [{...}, {...}, {...}]
+    rules_by_state = {}
+    for i, rule_def in enumerate(rule_defs, 1):
+        rule_dict = rule_def.get('rule', {})
+        if 'build_state' in rule_dict:
```

Comment:
This if-else-block could be rewritten as:
```python
build_state  = rule_dict.get('rule', {}).pop('build_state', conf.build_state)
```

Reply:
:+1:  Thanks!